### DISCUSSION TITLE
move away from transifex

### DISCUSSION THREAD

**Discussion by xinomilo** [searx/searx#2552] [Ideas]
well, just wanted to point out that transifex is proprietary software. 
https://github.com/transifex/transifex/issues/206#issuecomment-15243207
personally, i'd rather not work on that any more. dont know how other searx translators feel about that (?) 
would suggest weblate, but dont know any public instances other than https://hosted.weblate.org/

**Comment by Cqoicebordel** [comment]
I personnally don't care.

I'm commited to open source software, but when it comes to handling data that are not sensitive at all, I rather want to use the right tool than limiting myself to OSS.

**Reply by comradekingu** [reply]
If you actually use it, you will find that Weblate has more than one UI; all of which are more capable than the one on Transifex.
The "comfort" of Transifex is not being able to see more languages, hidden context, no space to translate, suggestions that are deleted, a broken messaging system, not enough checks, and so on.

**Comment by ghost** [comment]
I agree that hosting the translations on hosted.weblate.org would be better than on Transifex. However, even if this isn't changed, you can easily use free software like poEdit to translate locally, not needing any web services, free or proprietary, at all.

**Reply by comradekingu** [reply]
You aren't agreeing with anyone there. Pointing out "Transifex is proprietary software" I take to mean "move away from" as going with a libre alternative. Weblate is just as libre as a service, it is just not limited to it.
There are also plenty of spying trackers on Transifex, and egregious terms atop that. Not the same at all.

**Comment by dimqua** [comment]
What about [Zanata](https://translate.zanata.org/)? And, btw, GitHub is also a proprietary service.

**Reply by comradekingu** [reply]
RedHat has halted development on Zanata almost completely, in favour of Weblate.

**Comment by xinomilo** [comment]
well in any case i'll close this issue, since it doesn't seem people mind using transifex (or/and github). personally, i already quit transifex, so no more translations there for me. probably github in near future, as soon as there is a tested nice free alternative (and also not sexist one)

**Reply by comradekingu** [reply]
4 people are for moving, two people are indifferent, and one person against.
Since 2015 nothing happened on Transifex beyond a basic regexp search, and Weblate greatly improved.

**Comment by GreenLunar** [comment]
Ever since they became proprietary. I have seen this argument rising up in several projects already.

CopyQ, Exaile, Money Manager EX and even Gammu are currently being translated at https://hosted.weblate.org/ using [Weblate](http://weblate.org/) which provides integration with Git.

Please DO move away from the translation platform in question.

**Comment by GreenLunar** [comment]
BUMP!

**Comment by Cqoicebordel** [comment]
Changing platform will take time for the devs away of developping searx, when changing platform add zero advantage.
And beyond that, the teams of translators are already registered at transifex. So It would be a loss to move away from them.
I said I didn't care, but now, I think it's a bad idea to move away from transifex.

And beyond that, you can use a fully open source stack to contribute to the translations.

**Reply by comradekingu** [reply]
Not true at all. Quality would increase dramatically, and if you used the various alternatives, as much would have been obvious.
They are not all the same.

**Comment by andras-tim** [comment]
**For note:**
I'm find another web based translator tool, the [poeditor.com](http://poeditor.com).
This is free for open source projects (https://poeditor.com/help/faq#What-are-Open-Source-projects)

In other hand, I'm full agree with @Cqoicebordel

**Comment by ghost** [comment]
That's just another proprietary web 'service' so it doesn't change the situation. Unless they've hidden their licensing and code somewhere, but projects that do that are not worth looking at anyway because they have no interest in freedom.

**Comment by GreenLunar** [comment]
I refuse to keep the translation platform in question.

When more and more projects would show contempt for Proprietary Services that used to be also available as Free Software, they might reverse back.  GNU MediaGoblin has moved away from the translation platform in question to Pootle.  Also, we can gradually migrate translators and translators can even login to Weblate with GitHub and other social network accounts.

P.S. I have agreed to register to the translation platform in question because it WAS available as Free Software.

VOTE WITH YOUR WALLETS! VOTE FOR FREEDOM!

**Comment by Cqoicebordel** [comment]
There is no ads, and nobody here is donating money to Transifex.
When and if a paid platform will be considered, of course an open source one will be used.

Oh, and Transifex isn't a bad word, you can use it.

For migrating translators, they choose what they do. And some of them may not be on github, and some of them are not devs. And so, some of them will be lost when changing platform.

Searx's code and translations are still open source, no matter what. Even if transifex is closed source, and even if GitHub is closed source, and even if the routers at your FAI use closed source firmware.

But nobody forces you to use, or to translate, or to contribute. You are welcome to do it, but nobody forces you.

**Reply by comradekingu** [reply]
There are provisions for ads, and your info is getting sold. Read the terms and services.
Look at what scrips are being loaded.
The rest is just beside the point.
What happens to be the case is at least 3 translators here refuse to use Transifex.
Those aren't hypothetical.
Further, Transifex isn't free for libre software, whereas Weblate is. Being in that negotiation is not a place you want to be, because their pricing is steep. This has happened many times to libre software projects.

**Comment by GreenLunar** [comment]
> Oh, and "the translation platform in question" isn't a bad word, you can use it.

Only if they pay me, every time I do ;-)

> For migrating translators, they choose what they do. And some of them may not be on github, and some of them are not devs. And so, some of them will be lost when changing platform.

Translators can even login to Weblate with GitHub and other social network accounts, such as F and G.

> Searx's code and translations are still open source, no matter what. Even if transifex is closed source, and even if GitHub is closed source, and even if the routers at your FAI use closed source firmware.

I use DD-WRT.  The translation platform in question was available as Free Software.

**Comment by Cqoicebordel** [comment]
> Translators can even login to Weblate with GitHub and other social network accounts, such as F and G.

My point was that a lot a them won't even know that we switched platform.
Oh, and BTW, you prefer using an open source platform but recommand using closed source credentials ? That's weird ;)

> I use DD-WRT. The translation platform in question was available as Free Software.

We are moving away from the subject, but I was talking about your ISP, not you. I'm sorry, FAI is a french acronym. And I'm sure your ISP use closed source Cisco firmwares ;)

**Comment by GreenLunar** [comment]
> Oh, and BTW, you prefer using an open source platform but recommand using closed source credentials ? That's weird ;)

Did I recommend?

I am not willing to  continue this thread with you.

I think that migrating to Pootle or Weblate (even better) would be good.

**Comment by GreenLunar** [comment]
Comment by nijel

> Currently this feature is available only for commercial (non public) translations. This might change once nijel/weblate/issues/516 is implemented.

Do not use Weblate, yet.

When restricting access would be possible, I will ask, again, to migrate to Weblate.

**Comment by comradekingu** [comment]
@GreenLunar The issue is closed https://github.com/WeblateOrg/weblate/issues/516
I think this discussion fell short of addressing why the translation platform doesn't live up to what the product sells itself on. Moreover the 2021 situation is Transifex hasn't improved, whereas Weblate surpassed it. Was hoping to work on this, and was quite surprised to find searx uses Transifex. It spies on users left and right, and while it is true a lot of effort could be put in mitigation, that doesn't scale, and it isn't even a good tool.

**Comment by GreenLunar** [comment]
Hi, please STAY on Transifex or move to your other platforms AS LONG AS those platforms are on your own private servers (your = you = searx members).

It's not just me, but there is a phenomenon in which people the world over who are unemployed have a habit to corrupt translations. Moving to weblate on under weblate's servers would open that risk again as they don't have permission restriction system and no control management as to who's modifying what.

Stay, as long as the format geberated (PO, MO files) is open and we have no legal restrictions on our own work, STAY.

**Comment by comradekingu** [comment]
@GreenLunar 
However laudable, what exactly do you intend to remedy by moving to a server of your own? Transifex doesn't offer that _at all_. Staying there falls short of attaining this goal no matter the reasoning as to why that could be better.

Restricting user freedom isn't suddenly OK just because there is a system of hierarchy in place. Nor is that lost, but you fail to account for missing usability and the quality of the end-product entirely.

Where exactly does this problem of unemployed translators corrupting translations occur? If you mean bad quality translations, Transifex is chiefly the system producing it, as the home of drive-by translators at large. The point of moving isn't to find the system of the most stringent restrictions, but rather getting away from the supposed need for it. Hence the discussion. Transifex isn't good.

In my professional capacity as a translator and one having used every platform extensively, Transifex is not solving any problems of corruption. It holds projects ransom to their extortionate pricing once it feels the need to, and it uses the libre software community to do its bidding while saying it will still help it. What it actually did is reducing its offering to non-commercial projects.

The problem isn't even unemployed people, which is what most of your translators will be anyhow. You are fundamentally either missing or refusing to deal with the issue. The problem is Transifex  plaguing good translators and not solving the problem of ensuring quality at any level. The base quality is the increasing quality of more people helping, in turn for equally increased chances of inconsistency occurring. Run this experiment on the usability of Transifex between enough libre software projects and you exhaust even the best translators beyond ability, for no good reason.

---

The control seeking role-based system of Transifex is broken as a means of ensuring control of quality, because it means spanning the level of control and personal vetting to the limits of fewer people with different hats labeled "language coordinators", or more likely, "project admins". You don't go down this road without going all the way and meeting quality at the door for every string.

Having picked your horses, you then end up with a first-taker problem of reviewers being those first seeking to become one, and then gauging quality there (or beyond that) by someone who doesn't speak the language. Never not the case on Transifex, but do enlighten me. Or do you hypothetically intend to find the best language coordinators for every language as an extended problem to the one you already have (and aren't doing anything in your control to mitigate effectively anytime soon)? You are at the mercy of a broken, closed-source system blaming other people for unwarranted reason.

If you lock reviewed translations as is often the case on Transifex that cements the problem rather than solve it. Voting on a 2vote-to-include basis is also something that trashes consistency. You remove the ability of anyone trying to fix a whole resource, by relying on everyone explaining everything all the time in comment messages, or a non-threaded communication system nobody reads. What this does to prevent malicious intent from accounts registered two days ago is ensuring said person also has an account registered yesterday to enforce what is indistinguishable from malice otherwise.

Transifex quality scales to the proficiency of any one translator arriving at any level of consistency, and that part is so much harder than it needs to be. Harder still with roles.

---

Self-hosted Weblate has every restriction and then some, but in your misunderstanding of it, you don't know what you are missing. Understanding Hosted Weblate also has a system of roles of restrictions, isn't even where your confusion arose.

Quality stems from overview, communication, and vested interest. It is a product of community. Transifex doesn't do better in any capacity there.  Next to nothing is open, communication is lackluster, and the UI is a pain. On the contrary being able to change everything everywhere produces better quality, evidently. It also creates vested interest in not getting banned for what is a system of relational quality. How do you get that sort of overview?

If you start with every translation removed from view of the public, and translators at large, that is the problem you have. You will never replace the insight betwixt those people by way of personal management. As somewhat of an irony, you can look up how many projects "kingu" manages if you are logged in. (Not that I have used since they changed their terms following EU-GDPR) The functionality of Transifex is however a continual and ever-returning topic among translators, so I know _exactly_ how it is still broken. It also removes itself from developers, who don't want to use it, as much as it removes the source repo from the translation of it.

Weblate isn't perfect as there are things missing to help its model, but what it doesn't have is the type of broken stopgap functionality that Transifex employs everywhere.

---

Just because it doesn't have visual ads doesn't mean Transifex isn't selling your data. (@Cqoicebordel A full-stack of employed libre software text-editing means having to employ it, and doesn't even begin to deal with any of the bigger problems)
As for legal restrictions, read the [Transifex terms of service](https://www.transifex.com/legal/terms/). Having to be 18 to accept it all is only the first legal restriction to overcome. Then there is the DMCA, Acceptable Use, Cookies policy, and the Security Policy.
Then there is every equivalent of all the third-party scripts it uses. Then there is international privacy, consumer and trade law for no good reason.

Not only is the Transifex EULA a spaghetti legalese mess of legal restrictions and ramifications that spans between these documents, it also makes serious inroads on the freedom of translators and the product of their work, which is a better "we" at that. **Valuing the freedom of Searx translators differently is only an inconsistency in how Searx values user freedom.**

Your initial premise and the 3 times you wrote "stay" are thus voided.

TL;DR A system producing translations of poorer quality isn't solved by arbitrary restrictions at one end any more than the unrestricted nature of the file format you end up with at the other being anything to lean on.

**Reply by GreenLunar** [reply]
I'm for a self hosted instance or any other open-source alternative, as long as searx team maintains control.

The unemployed translators is a term I use for people who don't bother asking original translators (active or not) and dumping text in order to have their name in PO file, not to contribute* (and consequently mentioning that in CV) - this phenomenon varies from language to language.

* Resulting in a corrupted and unappealing translation.

**Reply by comradekingu** [reply]
That means moving.

**Comment by comradekingu** [comment]
@aurora-vasiliev Why are you voting this down?
